analyzer/codechecker_analyzer/cmd/fixit.py

```diff
@@ -0,0 +1,242 @@
+# -------------------------------------------------------------------------
+#
+#  Part of the CodeChecker project, under the Apache License v2.0 with
+#  LLVM Exceptions. See LICENSE for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# -------------------------------------------------------------------------
+"""
+Defines the CodeChecker action for applying fixits.
+"""
+
+import argparse
+import json
+import os
+import re
+import subprocess
+import sys
+import tempfile
+import yaml
+
+from codechecker_analyzer import analyzer_context
+from codechecker_common import arg, logger
+
+LOG = logger.get_logger('system')
+
+
+def get_argparser_ctor_args():
+    """
+    This method returns a dict containing the kwargs for constructing an
+    argparse.ArgumentParser (either directly or as a subparser).
+    """
+
+    return {
+        'prog': 'CodeChecker fixit',
+        'formatter_class': arg.RawDescriptionDefaultHelpFormatter,
+        'description': """
+Some analyzers may suggest some automatic bugfixes. Most of the times these are
+style issues which can be fixed easily. This command handles the listing and
+application of these automatic fixes.
+
+Besides the provided filter options you can pipe the JSON format output of
+"CodeChecker cmd diff" command to apply automatic fixes only for new reports:
+CodeChecker cmd diff -b dir1 -n dir2 -o json --new | CodeChecker fixit dir2""",
```

Comment:
I tried to run the following command but I got an exception:
```
CodeChecker cmd diff -b ~/projects/simple/reports -n simple --new -o json | CodeChecker fixit

[INFO 2020-06-30 15:19] - Compared local report directory(s) ~/projects/simple/reports and remote run(s) simple (matching: simple).
Exception ignored in: <_io.TextIOWrapper name='<stdout>' mode='w' encoding='UTF-8'>
BrokenPipeError: [Errno 32] Broken pipe
```

I also tried to run the following command but only the help message of the CodeChecker fixit command was shown:
```sh
CodeChecker cmd diff -n ~/projects/simple/reports -b simple --new -o json | CodeChecker fixit ~/projects/simple/reports

[INFO 2020-06-30 15:22] - Checking for local valid sessions.
[INFO 2020-06-30 15:22] - Compared remote run(s) simple (matching: simple) and local report directory(s) ~/projects/simple/reports
usage: CodeChecker [-h] {fixit} ...
```

Reply:
I fixed the second issue, however, I couldn't find a solution to the first one yet.